src/aggregate/aggregate_request.c

```diff
@@ -114,42 +114,30 @@ static int parseCursorSettings(AREQ *req, ArgsCursor *ac, QueryError *status) {
   return REDISMODULE_OK;
 }
 
-#define ARG_HANDLED 1
-#define ARG_ERROR -1
-#define ARG_UNKNOWN 0
+static int parseRequiredFields(AREQ *req, ArgsCursor *ac, QueryError *status){
 
-static int parseParams (AREQ *req, ArgsCursor *ac, QueryError *status) {
-  ArgsCursor paramsArgs = {0};
-  int rv = AC_GetVarArgs(ac, &paramsArgs);
+  ArgsCursor args = {0};
+  int rv = AC_GetVarArgs(ac, &args);
   if (rv != AC_OK) {
-    QERR_MKBADARGS_AC(status, "PARAMS", rv);
-    return REDISMODULE_ERR;
-  }
-  if (req->searchopts.params) {
-    QueryError_SetError(status, QUERY_EADDARGS,"Multiple PARAMS are not allowed. Parameters can be defined only once");
-    return REDISMODULE_ERR;
-  }
-  if (paramsArgs.argc == 0 || paramsArgs.argc % 2) {
-    QueryError_SetError(status, QUERY_EADDARGS,"Parameters must be specified in PARAM VALUE pairs");
+    QERR_MKBADARGS_AC(status, "_REQUIRED_FIELDS", rv);
     return REDISMODULE_ERR;
   }
 
-  dict *params = Param_DictCreate();
-  size_t value_len;
-  while (!AC_IsAtEnd(&paramsArgs)) {
-    const char *param = AC_GetStringNC(&paramsArgs, NULL);
-    const char *value = AC_GetStringNC(&paramsArgs, &value_len);
-    // FIXME: Validate param is [a-zA-Z][a-zA-z_\-:0-9]*
-    if (DICT_ERR == Param_DictAdd(params, param, value, value_len, status)) {
-      Param_DictFree(params);
-      return REDISMODULE_ERR;
-    }
+  int requiredFieldNum = AC_NumArgs(&args);
+  const char** requiredFields = array_new(const char*, requiredFieldNum);
+  for(size_t i=0; i < requiredFieldNum; i++) {
+    requiredFields = array_append(requiredFields, AC_GetStringNC(&args, NULL));
```

Comment:
You do not check input correctness here, you just continue to read until `requiredFieldNum` reached, do you check it elsewhere?

Reply:
`AC_GetVarArgs` checks about the var args array
added a check for NULL string
thanks

Reply:
Look at the last comment I posted.